### ISSUE TITLE
Update Meeting Info button

### ISSUE THREAD

**Issue by Grateful-Viv** [code4recovery/tsml-ui#127]
Right now the biggest issues I see with switching to the ui front end is the update meeting link (see https://github.com/code4recovery/tsml-ui/issues/114). I don’t know what an airtable is and how the data for Wordpress is different. But I do like how the aa San Francisco solution works. I’d also be willing to investigate Wordpress free (and even premium) forms as long as the link to the meeting could be passed and display the meeting title (rather than URL) to the end user. 

The URL, I’ve noticed is a bit weird, depending upon how you got there. For example, https://sdi-staging.sussexaa.org/sdi-meetings/?region=lewes&meeting=bb-study-group and https://sdi-staging.sussexaa.org/sdi-meetings/?region=lewes&type=active&meeting=bb-study-group are the same meeting. Ideally, it should be https://sdi-staging.sussexaa.org/sdi-meetings/bb-study-group. 

The other issue has to do with wanting different information on the meeting detail from the meeting list view which are handled by the widget plugins. That isn’t a showstopper, however. I would just modify the html on the page to make sense for both views.

**Comment by joshreisner** [comment]
Oops, I had accidentally enabled the button for all meetings. It's supposed to be enabled only when there is an email address to send to. This will be the case on TSML once my [pull request](https://github.com/code4recovery/12-step-meeting-list/pull/341 ) is merged and the new version ships.

Regarding the URL, we could probably strip out the search terms that got the user there, making it  https://sdi-staging.sussexaa.org/sdi-meetings/?meeting=bb-study-group

We could also try putting some meeting info into the default body of the message, but this is more likely to get edited by the user. Perhaps to start I'll just try adding name, time, location.

**Comment by Grateful-Viv** [comment]
Would it be possible to have the url without the “?meeting=“ ?

this https://sdi-staging.sussexaa.org/meetings/wake-up-call-24-hours-day-indoors/ instead of https://sdi-staging.sussexaa.org/sdi-meetings/?meeting=wake-up-call-24-hours-day

**Comment by joshreisner** [comment]
Maybe, with a deep enough integration with TSML. We'll certainly have to have a system where if people visit the first URL they are redirected to the second URL, but it could be eventually that we can add React Router and achieve that URL structure. It's a big lift and would depend on a lot of things so I'm not sure if it'll work.

It sounds like you're going in a different direction for meeting feedback, right? If so I'm going to close this ticket and not work on formatting the email.

**Comment by StudioLisa** [comment]
What's the objection here - that the subject line is hard to parse for the recipient of the email generated or difficulty in that data getting automatically entered into a database for maintenance or is it the difference between using the web-form front end vs. calling the user's email client?

**Comment by joshreisner** [comment]
I think for Viv it's mainly that it's more work for the her (as the admin) because she receives the email, clicks the subject, figures out which meeting it is on the backend, then edits that meeting.

The WordPress form system was a little cleaner and it automatically appended a link Viv could use to click and edit the meeting directly.

What might be a little better would be:

Subject: Meeting Feedback Form

Message:

Part at the top where the user can fill some stuff out

---- (this line and below automatically added)

Public URL: https://mywebsite.org/meetings?meeting=mymeeting
Edit URL: https://mywebsite.org/wp-admin/edit-meeting.php?id=980234

**Comment by Grateful-Viv** [comment]
I wasn’t thinking just about me with the update message request. 

At some point I’m going to turn this over to other volunteers, and they will have a difficult time negotiating which meeting is in need of updating. 

I like what you’re proposing, Josh.

**Comment by StudioLisa** [comment]
Got it. So as I understand it, the UX issue here is it’s hard for users to know what listing needs attention. 

The form (WordPress method) provided some way other than simply the URL for the listing and whatever it was that it provided was easier for users to address than simply the URL of the listing.

There are lots of UX issues with calling email clients anyway (such as users who only use webmail and never configured browser to know what to do with ‘mailto’s’ but meanwhile, what about putting the generating URL info into the body of the email vs. the subject field. Will this work?

***@***.*** ***@***.***>?subject=OutdatedInfo&body=<a href>whateverURL</a>…

As you can see here, the subject field alerts the user to some idea of the issue and the reference info needed (the generating URL) is pre-populated into the message body (formerly in the subject field) where users reporting can expand upon whatever needs fixing.

Does each listing have a unique ID? If so, that opens up lots of possibilities that might make such tasks even easier…
Which someone already has probably thought of or is working on. 

Hope this was helpful to someone.

Thanks for all you do.

LR

**Comment by joshreisner** [comment]
Ok! I've made the changes I proposed. @StudioLisa I think we're talking about the same thing with moving the URL to the body of the email. You can see a demo now at:

https://react.meetingguide.org/demo.html?meeting=cssly

Edit URL is not a variable that TSML provides yet, but I've added an issue here: https://github.com/code4recovery/12-step-meeting-list/issues/438

(You can see edit URL working at the example above because the data source is a Google Sheet.)

**Comment by Grateful-Viv** [comment]
@joshreisner, I see this has been updated to now open an email with the meeting URL in the body of the email and the subject Meeting Feedback: Meeting Name. Very nice!

Would it be possible to have the email open in a new browser window?

**Comment by StudioLisa** [comment]
That behavior is a behavior controlled by the user. Example: You would go into your browser settings to tell your browser to automatically open content in a separate window, etc. For obvious safety and security reasons, developers cannot, without a hell of a lot of challenges, take over one’s personal browser settings. 

LR

**Comment by StudioLisa** [comment]
...Also, forgot to add… if the user is using an email client vs. webmail, the operating system and their personal configuration of their environment has control over that… In other words, remember: “Email” does not necessarily mean webmail - it means email, period and an email can be created either via a web app (‘webmail’) or via a native running app (example: Apple Mail or MS Outlook). Yes, a pop up can be programmed to open a webmail window if the user uses webmail but that has other UX problems (ex., user has configured a browser to not display pop up’s).

This current design change was arrived at as the best given the current constraints all around. 

LR
